File: src/dataset/c/isogram_Exercism_vlzware/gemini/GeminiFlash_isogram_v1.c
#include <stdbool.h>
#include <string.h>
#include <ctype.h>

#define CCOUNT ('z' - 'a' + 1)

bool is_isogram(const char phrase[])
{
    if (!phrase) return true;

    bool letters[CCOUNT] = {false};
    const char *p = phrase;
    int k;
    char c;

    while (*p) {
        c = tolower(*p);

        if (c == ' ' || c == '-') {
            p++;
            continue;
        }

        if (!isalpha(c)) {
            return false;
        }

        k = c - 'a';
        if (letters[k]) {
            return false;
        }

        letters[k] = true;
        p++;
    }

    return true;
}

File: src/dataset/typescript/isogram_Exercism-typescript-FilipeCerejo/gemini/GeminiFlash_isogram_v1.ts
export function isIsogram(phrase: string): boolean {
  const letters = phrase.toLowerCase().replace(/[^a-z]/g, '');
  const letterSet = new Set<string>();

  for (let i = 0; i < letters.length; i++) {
    const char = letters[i];
    if (letterSet.has(char)) {
      return false;
    }
    letterSet.add(char);
  }

  return true;
}